thriftbp/thrifttest/server_example_test.go

```diff
@@ -0,0 +1,60 @@
+package thrifttest_test
+
+import (
+	"context"
+	"testing"
+
+	baseplatethrift "github.com/reddit/baseplate.go/internal/gen-go/reddit/baseplate"
+	"github.com/reddit/baseplate.go/secrets"
+	"github.com/reddit/baseplate.go/thriftbp/thrifttest"
+)
+
+type mockedBaseplateService struct {
+	Fail bool
+	Err  error
+}
+
+func (srv mockedBaseplateService) IsHealthy(ctx context.Context, req *baseplatethrift.IsHealthyRequest) (r bool, err error) {
+	return !srv.Fail, srv.Err
+}
+
+// In real test this function needs to be named TestService instead,
+// but doing that will break this example.
+func ServiceTest(t *testing.T) {
+	// Initialize this properly in a real test
+	var store *secrets.Store
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	processor := baseplatethrift.NewBaseplateServiceV2Processor(mockedBaseplateService{})
+	server, err := thrifttest.NewBaseplateServer(thrifttest.ServerConfig{
+		Processor:   processor,
+		SecretStore: store,
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+	// cancelling the context will close the server.
+	server.Start(ctx)
+
+	client := baseplatethrift.NewBaseplateServiceV2Client(server.ClientPool.TClient())
+	success, err := client.IsHealthy(ctx, &baseplatethrift.IsHealthyRequest{
+		Probe: baseplatethrift.IsHealthyProbePtr(baseplatethrift.IsHealthyProbe_READINESS),
+	})
+
+	if err != nil {
+		t.Errorf("expected no error, got %v", err)
+	}
+
+	const expected = true
+	if success != expected {
```

Comment:
Any thoughts on standardizing this to `got` & `want`?

Reply:
I personally usually use `expected` vs. `actual` but I guess `got` vs. `want` is shorter.